common/src/main/java/io/netty/util/internal/MpscLinkedQueue.java

```diff
@@ -197,22 +211,16 @@ public boolean contains(Object o) {
     @Override
     public Iterator<E> iterator() {
         return new Iterator<E>() {
-            private MpscLinkedQueueNode<E> node = peekNode();
+            private final Iterator<E> it = toList(16).iterator();
```

Comment:
This would be a candidate for https://github.com/netty/netty/pull/3684/files#r29030165